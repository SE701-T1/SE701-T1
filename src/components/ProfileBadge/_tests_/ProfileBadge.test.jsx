import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import ProfileBage from '../ProfileBadge';

describe('Render ProfileBadge', () => {
  it('Render Active ProfileBadge', () => {
    render(<ProfileBage active name="Bob" />);
    expect(screen.getByTestId('active'));
  });
  it('Render Offline ProfileBadge', () => {
    render(<ProfileBage active={false} name="Bob" />);
    expect(screen.getByTestId('offline'));
    expect(screen.getByTestId('Bob'));
  });

  // snapshot test
  it('produces correct snapshot of the ProfileBadge', () => {
    const tree = renderer.create(<ProfileBage active name="Bob" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
