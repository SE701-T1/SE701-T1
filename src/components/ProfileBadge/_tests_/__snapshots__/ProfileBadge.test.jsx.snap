// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ProfileBadge produces correct snapshot of the ProfileBadge 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-6ch1n3-MuiContainer-root"
>
  <span
    className="MuiBadge-root css-1srwgqi-MuiBadge-root"
    data-testid="active"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1pl8iwh-MuiAvatar-root"
      data-testid="Bob"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
        data-testid="PersonIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <span
      className="MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight css-ank3r8-MuiBadge-badge"
    >
      
    </span>
  </span>
</div>
`;
